Intermittent JS test failure

From https://travis-ci.org/mitodl/micromasters/jobs/207507688
    
    
    [92m [0m[32m 896 passing[0m[90m (1m)[0m
    
    [31m  2 failing[0m
    
    
    
    [0m  1) DashboardPage order receipt and cancellation pages course pricing "before each" hook:
    
    [0m[31m     Uncaught [0m[90m
    
      NotFoundError
    
          at HTMLBodyElementImpl.removeChild (node_modules/jsdom/lib/jsdom/living/nodes/Node-impl.js:315:13)
    
          at HTMLBodyElement.removeChild (node_modules/jsdom/lib/jsdom/living/generated/Node.js:160:57)
    
          at RenderToLayer.unrenderLayer (node_modules/material-ui/internal/RenderToLayer.js:109:21)
    
          at RenderToLayer.componentWillUnmount (node_modules/material-ui/internal/RenderToLayer.js:85:12)
    
          at node_modules/react-dom/lib/ReactCompositeComponent.js:409:25
    
          at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)
    
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:408:11)
    
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
    
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:418:23)
    
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
    
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:418:23)
    
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
    
          at Object.ReactChildReconciler.unmountChildren (node_modules/react-dom/lib/ReactChildReconciler.js:146:25)
    
          at ReactDOMComponent.ReactMultiChild.Mixin.unmountChildren (node_modules/react-dom/lib/ReactMultiChild.js:373:28)
    
          at ReactDOMComponent.Mixin.unmountComponent (node_modules/react-dom/lib/ReactDOMComponent.js:980:10)
    
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
    
          at Object.ReactChildReconciler.unmountChildren (node_modules/react-dom/lib/ReactChildReconciler.js:146:25)
    
          at ReactDOMComponent.ReactMultiChild.Mixin.unmountChildren (node_modules/react-dom/lib/ReactMultiChild.js:373:28)
    
          at ReactDOMComponent.Mixin.unmountComponent (node_modules/react-dom/lib/ReactDOMComponent.js:980:10)
    
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
    
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:130:25)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
    
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
    
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
    
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
    
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
    
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
    
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
    
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:109:25)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
    
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
    
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
    
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
    
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
    
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
    
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
    
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:109:25)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
    
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
    
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
    
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
    
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
    
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
    
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
    
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
    
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
    
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
    
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
    
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
    
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
    
          at ReactCompositeComponent.performUpdateIfNecessary (node_modules/react-dom/lib/ReactCompositeComponent.js:561:12)
    
          at Object.ReactReconciler.performUpdateIfNecessary (node_modules/react-dom/lib/ReactReconciler.js:157:22)
    
          at runBatchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:150:21)
    
          at ReactReconcileTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:140:20)
    
          at ReactUpdatesFlushTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:140:20)
    
          at ReactUpdatesFlushTransaction._assign.perform (node_modules/react-dom/lib/ReactUpdates.js:89:32)
    
          at Object.flushBatchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:172:19)
    
          at ReactDefaultBatchingStrategyTransaction.TransactionImpl.closeAll (node_modules/react-dom/lib/Transaction.js:206:25)
    
          at ReactDefaultBatchingStrategyTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:153:16)
    
          at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:62:26)
    
          at Object.enqueueUpdate (node_modules/react-dom/lib/ReactUpdates.js:200:22)
    
          at enqueueUpdate (node_modules/react-dom/lib/ReactUpdateQueue.js:24:16)
    
          at Object.ReactUpdateQueue.enqueueSetState (node_modules/react-dom/lib/ReactUpdateQueue.js:209:5)
    
          at Connect.ReactComponent.setState (node_modules/react/lib/ReactComponent.js:63:16)
    
          at Connect.onStateChange (node_modules/react-redux/lib/components/connectAdvanced.js:217:22)
    
          at Object.notify (node_modules/react-redux/lib/utils/Subscription.js:30:21)
    
          at Subscription.notifyNestedSubs (node_modules/react-redux/lib/utils/Subscription.js:65:20)
    
          at Connect.onStateChange (node_modules/react-redux/lib/components/connectAdvanced.js:210:30)
    
          at dispatch (node_modules/redux/lib/createStore.js:186:19)
    
          at node_modules/redux-asserts/lib/index.js:147:16
    
          at node_modules/redux-thunk/lib/index.js:14:16
    
          at dispatch (node_modules/redux/lib/applyMiddleware.js:45:18)
    
          at static/js/actions/dashboard.js:2:1266
    
          at node_modules/redux-thunk/lib/index.js:11:18
    
          at node_modules/redux-localstorage/lib/persistStateMiddleware.js:26:22
    
          at node_modules/redux-localstorage/lib/bufferActions.js:29:29
    
          at Timeout._onTimeout (static/js/containers/DashboardPage.js:1:74579)
    
    [0m
    
    [0m  2) DashboardPage "before each" hook for "refetches the dashboard after 3 seconds if 30 seconds has not passed":
    
    [0m[31m     TypeError: Attempted to wrap getDashboard which is already wrapped[0m[90m
    
          at checkWrappedMethod (node_modules/sinon/lib/sinon/util/core.js:81:29)
    
          at Object.wrapMethod (node_modules/sinon/lib/sinon/util/core.js:121:21)
    
          at Object.stub (node_modules/sinon/lib/sinon/stub.js:67:26)
    
          at Object.stub (node_modules/sinon/lib/sinon/collection.js:117:44)
    
          at new IntegrationTestHelper (static/js/util/integration_test_helper.js:3:395)
    
          at Context.<anonymous> (static/js/containers/DashboardPage_test.js:1:48758)
    
      --------------
    
      Error: Stack Trace for original
    
          at Object.wrapMethod (node_modules/sinon/lib/sinon/util/core.js:140:34)
    
          at Object.stub (node_modules/sinon/lib/sinon/stub.js:67:26)
    
          at Object.stub (node_modules/sinon/lib/sinon/collection.js:117:44)
    
          at new IntegrationTestHelper (static/js/util/integration_test_helper.js:3:395)
    
          at Context.<anonymous> (static/js/containers/DashboardPage_test.js:1:48758)
    
    [0m

Here's one with correct line numbers:

        handles redeeming coupons
          ✓ with a successful fetch (193ms)
          ✓ with a failed fetch (152ms)
          1) without a race condition
        coupon messaging
          ✓ has coupon messaging for a program (123ms)
          2) has coupon messaging for a program
          course contact UI behavior
            3) shows the email composition dialog when a user has permission to contact a course team
            4) "before each" hook for "shows the email composition dialog when a user has permission to contact a course team"
    
      FinancialAidCalculator
    
    
      149 passing (10s)
      4 failing
    
      1) DashboardPage handles redeeming coupons without a race condition:
         Uncaught 
      NotFoundError
          at HTMLBodyElementImpl.removeChild (node_modules/jsdom/lib/jsdom/living/nodes/Node-impl.js:315:13)
          at HTMLBodyElement.removeChild (node_modules/jsdom/lib/jsdom/living/generated/Node.js:160:57)
          at RenderToLayer.unrenderLayer (node_modules/material-ui/internal/RenderToLayer.js:109:21)
          at RenderToLayer.componentWillUnmount (node_modules/material-ui/internal/RenderToLayer.js:85:12)
          at node_modules/react-dom/lib/ReactCompositeComponent.js:409:25
          at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:408:11)
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:418:23)
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
          at ReactCompositeComponent.unmountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:418:23)
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
          at Object.ReactChildReconciler.unmountChildren (node_modules/react-dom/lib/ReactChildReconciler.js:146:25)
          at ReactDOMComponent.ReactMultiChild.Mixin.unmountChildren (node_modules/react-dom/lib/ReactMultiChild.js:373:28)
          at ReactDOMComponent.Mixin.unmountComponent (node_modules/react-dom/lib/ReactDOMComponent.js:980:10)
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
          at Object.ReactChildReconciler.unmountChildren (node_modules/react-dom/lib/ReactChildReconciler.js:146:25)
          at ReactDOMComponent.ReactMultiChild.Mixin.unmountChildren (node_modules/react-dom/lib/ReactMultiChild.js:373:28)
          at ReactDOMComponent.Mixin.unmountComponent (node_modules/react-dom/lib/ReactDOMComponent.js:980:10)
          at Object.ReactReconciler.unmountComponent (node_modules/react-dom/lib/ReactReconciler.js:79:22)
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:130:25)
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:109:25)
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at Object.ReactChildReconciler.updateChildren (node_modules/react-dom/lib/ReactChildReconciler.js:109:25)
          at ReactDOMComponent.ReactMultiChild.Mixin._reconcilerUpdateChildren (node_modules/react-dom/lib/ReactMultiChild.js:208:32)
          at ReactDOMComponent.ReactMultiChild.Mixin._updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:312:31)
          at ReactDOMComponent.ReactMultiChild.Mixin.updateChildren (node_modules/react-dom/lib/ReactMultiChild.js:299:12)
          at ReactDOMComponent.Mixin._updateDOMChildren (node_modules/react-dom/lib/ReactDOMComponent.js:936:12)
          at ReactDOMComponent.Mixin.updateComponent (node_modules/react-dom/lib/ReactDOMComponent.js:754:10)
          at ReactDOMComponent.Mixin.receiveComponent (node_modules/react-dom/lib/ReactDOMComponent.js:716:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
          at ReactCompositeComponent.receiveComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:547:10)
          at Object.ReactReconciler.receiveComponent (node_modules/react-dom/lib/ReactReconciler.js:125:22)
          at ReactCompositeComponent._updateRenderedComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:754:23)
          at ReactCompositeComponent._performComponentUpdate (node_modules/react-dom/lib/ReactCompositeComponent.js:724:10)
          at ReactCompositeComponent.updateComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:645:12)
          at ReactCompositeComponent.performUpdateIfNecessary (node_modules/react-dom/lib/ReactCompositeComponent.js:561:12)
          at Object.ReactReconciler.performUpdateIfNecessary (node_modules/react-dom/lib/ReactReconciler.js:157:22)
          at runBatchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:150:21)
          at ReactReconcileTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:140:20)
          at ReactUpdatesFlushTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:140:20)
          at ReactUpdatesFlushTransaction._assign.perform (node_modules/react-dom/lib/ReactUpdates.js:89:32)
          at Object.flushBatchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:172:19)
          at ReactDefaultBatchingStrategyTransaction.TransactionImpl.closeAll (node_modules/react-dom/lib/Transaction.js:206:25)
          at ReactDefaultBatchingStrategyTransaction.TransactionImpl.perform (node_modules/react-dom/lib/Transaction.js:153:16)
          at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:62:26)
          at Object.enqueueUpdate (node_modules/react-dom/lib/ReactUpdates.js:200:22)
          at enqueueUpdate (node_modules/react-dom/lib/ReactUpdateQueue.js:24:16)
          at Object.ReactUpdateQueue.enqueueSetState (node_modules/react-dom/lib/ReactUpdateQueue.js:209:5)
          at Connect.ReactComponent.setState (node_modules/react/lib/ReactComponent.js:63:16)
          at Connect.onStateChange (node_modules/react-redux/lib/components/connectAdvanced.js:217:22)
          at Object.notify (node_modules/react-redux/lib/utils/Subscription.js:30:21)
          at Subscription.notifyNestedSubs (node_modules/react-redux/lib/utils/Subscription.js:65:20)
          at Connect.onStateChange (node_modules/react-redux/lib/components/connectAdvanced.js:210:30)
          at dispatch (node_modules/redux/lib/createStore.js:186:19)
          at node_modules/redux-asserts/lib/index.js:147:16
          at node_modules/redux-thunk/lib/index.js:14:16
          at dispatch (node_modules/redux/lib/applyMiddleware.js:45:18)
          at static/js/actions/dashboard.js:29:5
          at node_modules/redux-thunk/lib/index.js:11:18
          at node_modules/redux-localstorage/lib/persistStateMiddleware.js:26:22
          at node_modules/redux-localstorage/lib/bufferActions.js:29:29
          at Timeout._onTimeout (static/js/containers/DashboardPage.js:235:11)
    
      2) DashboardPage coupon messaging has coupon messaging for a program:
         AssertionError: expected null to deeply equal { Object (amount, amount_type, ...) }
          at Function.assert.deepEqual (node_modules/chai/lib/chai/interface/assert.js:216:32)
          at static/js/containers/DashboardPage_test.js:456:16
          at run (node_modules/core-js/modules/es6.promise.js:87:22)
          at node_modules/core-js/modules/es6.promise.js:100:28
          at flush (node_modules/core-js/modules/_microtask.js:18:9)
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
    
      3) DashboardPage course contact UI behavior shows the email composition dialog when a user has permission to contact a course team:
         TypeError: Cannot read property '0' of undefined
          at Context.<anonymous> (static/js/containers/DashboardPage_test.js:532:7)
    
      4) DashboardPage "before each" hook for "shows the email composition dialog when a user has permission to contact a course team":
         TypeError: Attempted to wrap getDashboard which is already wrapped
          at checkWrappedMethod (node_modules/sinon/lib/sinon/util/core.js:81:29)
          at Object.wrapMethod (node_modules/sinon/lib/sinon/util/core.js:121:21)
          at Object.stub (node_modules/sinon/lib/sinon/stub.js:67:26)
          at Object.stub (node_modules/sinon/lib/sinon/collection.js:117:44)
          at new IntegrationTestHelper (static/js/util/integration_test_helper.js:54:39)
          at Context.<anonymous> (static/js/containers/DashboardPage_test.js:100:14)
      --------------
      Error: Stack Trace for original
          at Object.wrapMethod (node_modules/sinon/lib/sinon/util/core.js:140:34)
          at Object.stub (node_modules/sinon/lib/sinon/stub.js:67:26)
          at Object.stub (node_modules/sinon/lib/sinon/collection.js:117:44)
          at new IntegrationTestHelper (static/js/util/integration_test_helper.js:54:39)
          at Context.<anonymous> (static/js/containers/DashboardPage_test.js:100:14)
    
    
    
    Tests failed, exiting with error 4...
    npm info lifecycle micromasters@0.0.0~test: Failed to exec test script
    npm ERR! Test failed.  See above for more details.

--- a/js_test.sh
+++ b/js_test.sh
@@ -48,6 +48,7 @@ fi
 if [[ $(
     cat "$TMP_FILE" |
     grep -v 'ignored, nothing could be mapped' |
+    grep -v "This browser doesn't support the \`onScroll\` event" |
     wc -l |
     awk '{print $1}'
     ) -ne 0 ]]  # is file empty?

--- a/static/js/containers/LearnerPage.js
+++ b/static/js/containers/LearnerPage.js
@@ -43,7 +43,7 @@ class LearnerPage extends React.Component<*, LearnerPageProps, *> {
 
   componentWillUnmount() {
     const { dispatch, params: { username } } = this.props;
-    if (SETTINGS.user.username !== username) {
+    if (!SETTINGS.user || SETTINGS.user.username !== username) {
       // don't erase the user's own profile from the state
       dispatch(clearProfile(username));
     }

--- a/static/js/global_init.js
+++ b/static/js/global_init.js
@@ -1,4 +1,6 @@
 // Define globals we would usually get from Django
+import ReactDOM from 'react-dom';
+
 const _createSettings = () => ({
   user: {
     username: "jane",
@@ -51,6 +53,10 @@ beforeEach(() => { // eslint-disable-line mocha/no-top-level-hooks
 
 // cleanup after each test run
 afterEach(function () { // eslint-disable-line mocha/no-top-level-hooks
+  let node = document.querySelector("#integration_test_div");
+  if (node) {
+    ReactDOM.unmountComponentAtNode(node);
+  }
   document.body.innerHTML = '';
   global.SETTINGS = _createSettings();
   window.localStorage.reset();

--- a/static/js/util/integration_test_helper.js
+++ b/static/js/util/integration_test_helper.js
@@ -110,6 +110,8 @@ export default class IntegrationTestHelper {
     return this.listenForActions(expectedTypes, () => {
       this.browserHistory.push(url);
       div = document.createElement("div");
+      div.setAttribute("id", "integration_test_div");
+      document.body.appendChild(div);
       wrapper = mount(
         <div>
           <DashboardRouter

